lib/resty/upstream/healthcheck.lua

```diff
@@ -435,6 +474,49 @@ local function get_lock(ctx)
     return true
 end
 
+local function set_all_down_flag(ctx, ppeers, bpeers)
+    local dict = ctx.dict
+    local u = ctx.upstream
+    local u_key = gen_upstream_key(u,"all_down")
+
+    local all_down, err = dict:get(u_key)
+    if not all_down then
+        errlog("Failed to get all down flag for upstream " .. u .. ". ", err)
+        return
+    end
+    
+    if all_down == 0 then
+        local result = true
```

Comment:
Please use more meaningful name for this variable. "result" carries no information about the meaning of its `true` or `false` values.

Reply:
Changed to "is_all_peers_down"